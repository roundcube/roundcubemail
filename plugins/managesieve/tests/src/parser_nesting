# Sieve Filter
# Erzeugt von Ingo (http://www.horde.org/ingo/) (30.09.2016, 16:02)

# Nested rules aren't supported and will be ignored (#5540)

require ["vacation", "regex"];

# Abwesenheit
if allof ( not exists ["list-help", "list-unsubscribe", "list-subscribe", "list-owner", "list-post", "list-archive", "list-id", "Mailing-List"],
  not header :is "Precedence" ["list", "bulk", "junk"],
  not header :matches "To" "Multiple recipients of*") {
    if header :regex "Received" "^.*(2016) (\\(.*\\) )?..:..:.. (\\(.*\\) )?(\\+|\\-)....( \\(.*\\))?$" {
    if header :regex "Received" "^.*(Oct) (\\(.*\\) )?.... (\\(.*\\) )?..:..:.. (\\(.*\\) )?(\\+|\\-)....( \\(.*\\))?$" {
    if header :regex "Received" "^.*([0 ]4|[0 ]5|[0 ]6|[0 ]7) (\\(.*\\) )?... (\\(.*\\) )?.... (\\(.*\\) )?..:..:.. (\\(.*\\) )?(\\+|\\-)....( \\(.*\\))?$" {
    vacation :days 7 :addresses "test@company.com" :subject "vacation" "blablabla";
}
}
}
}

# Ausgeschlossene Adressen
if address :is ["From", "Sender", "Resent-From"] "noreply@example.org"  {
    discard;
    stop;
}
