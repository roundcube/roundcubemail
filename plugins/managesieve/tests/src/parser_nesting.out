# Sieve Filter
# Erzeugt von Ingo (http://www.horde.org/ingo/) (30.09.2016, 16:02)
# Nested rules aren't supported and will be ignored (#5540)

# rule:[Ausgeschlossene Adressen]
if address :is ["From","Sender","Resent-From"] "noreply@example.org"
{
	discard;
	stop;
}
